packages/hardhat/test/integration-tests/periphery/controller-helper.ts

```diff
@@ -793,6 +793,210 @@ describe("Controller helper integration test", function () {
+      const ownerOfUniNFT = await (positionManager as INonfungiblePositionManager).ownerOf(tokenId); 
+      const position = await (positionManager as INonfungiblePositionManager).positions(tokenId)
+  
+      await (positionManager as INonfungiblePositionManager).connect(depositor).approve(positionManager.address, tokenId); 
+      const [amount0, amount1] = await (positionManager as INonfungiblePositionManager).connect(depositor).callStatic.decreaseLiquidity({
+        tokenId: tokenId,
+        liquidity: position.liquidity,
+        amount0Min: 0,
+        amount1Min: 0,
+        deadline: Math.floor(await getNow(ethers.provider) + 8640000),
+      })
+      const wPowerPerpAmountInLP = (isWethToken0) ? amount1 : amount0;
+      const wethAmountInLP = (isWethToken0) ? amount0 : amount1;
+
+      // console.log('position.tickLower', position.tickLower)
+      // console.log('position.tickUpper', position.tickUpper)
+      // console.log('ownerOfUniNFT', ownerOfUniNFT)
+      // console.log('depositor.address', depositor.address)
+      // console.log('wPowerPerpAmountInLP',wPowerPerpAmountInLP)
+      // console.log('wethAmountInLP',wethAmountInLP)
+      // console.log('vaultBefore.shortAmount',vaultBefore.shortAmount)
+      // console.log('vaultBefore.collateralAmount',vaultBefore.collateralAmount)
+      // console.log('vaultAfter.shortAmount',vaultAfter.shortAmount)
+      // console.log('vaultAfter.collateralAmount',vaultAfter.collateralAmount)
+      // console.log('depositorEthBalanceBefore',depositorEthBalanceBefore)
+      // console.log('depositorEthBalanceAfter',depositorEthBalanceAfter)
+      // console.log('collateralAmount',collateralAmount)
+      // console.log('gasSpent',gasSpent)
+      // console.log('depositorEthBalanceAfter',depositorEthBalanceAfter)
+      // console.log('collateralToLp',collateralToLp)
+      // console.log('mintWSqueethAmount',mintWSqueethAmount)
+      // console.log('tokenIndexBefore',tokenIndexBefore)
+      // console.log('tokenIndexAfter',tokenIndexAfter)
+
+      expect(position.tickLower === newTickLower).to.be.true
+      expect(position.tickUpper === newTickUpper).to.be.true
+      expect(ownerOfUniNFT === depositor.address).to.be.true
+      expect(tokenIndexAfter.sub(tokenIndexBefore).eq(BigNumber.from(1))).to.be.true
+      expect(vaultBefore.shortAmount.eq(BigNumber.from(0))).to.be.true
+      expect(vaultBefore.collateralAmount.eq(BigNumber.from(0))).to.be.true
+      expect(vaultAfter.collateralAmount.eq(BigNumber.from(0))).to.be.true
+      // One side LP - weth should be zero
+      expect(wethAmountInLP.sub(collateralToLp).abs().lte(10)).to.be.true
+      expect(wPowerPerpAmountInLP.eq(0)).to.be.true
+      expect((depositorEthBalanceBefore.sub(depositorEthBalanceAfter).sub(wethAmountInLP).sub(gasSpent)).abs().lte(1)).to.be.true
+      expect(wethAmountInLP.sub(collateralToLp).mul(one).div(collateralToLp).eq(0)).to.be.true
```

Comment:
similar comment, probably want to check that the wethAmountInLP is what is expected, not just conservation of weth

Reply:
Added `wethAmountInLP == collateralToLP` expect.